### ISSUE TITLE
Send headers doesn't pass RFC2616 Sec4.2:

### ISSUE THREAD

**Issue by thxer92** [phalcon/cphalcon#11131]
Hi,

I have a problem, and a "500 - Internal Server Error" created by Nginx, because one of header doesn't have a value.
According to the RFC2616: 
**Each header field consists of a name followed by a colon (":") and the field value.**

In phalcon/http/response/headers.zep, there is a send() function which do:

```zephir
if !empty value {
	header(header . ": " . value, true);
} else {
	header(header, true);
}
```
So if a passed header value is empty, an invalid header sent, which crash the webserver, and doesn't meet the requirements of RFC2616. 

Some value which is "empty" in PHP, and need to set as header value in some scenario:
* "" (an empty string)
* 0 (0 as an integer)
* 0.0 (0 as a float)
* "0" (0 as a string)
* NULL
* FALSE

**Comment by rianorie** [comment]
@thxer92 I'll take a look at it, but what header are you trying to set that's empty? And how does apache handle it?

Given the RFC not allowing an empty value would be the best.. So throwing an exception would technically be the best thing, but that will break backwards compatibility. So I'll probably end up sending out an empty value and maybe we tag it for Phalcon 3(!) for proper implementation.

@andresgutierrez thoughts?

**Comment by andresgutierrez** [comment]
What value can be used to avoid an empty value?

**Comment by rianorie** [comment]
I haven't checked but I suspect anything with a length above 0. I figured if it came to that simply outputting a textual null or something

**Comment by rianorie** [comment]
@thxer92 a valid header without a value as such: `header("HTTP/1.0 404 Not Found");`, so I can't simply append a null or void or whatever to it.. Can you elaborate on the header you're trying to send?

**Comment by andresgutierrez** [comment]
Fixed in 2.1.x branch
